### DISCUSSION TITLE
Sending objects forward and backward in layers

### DISCUSSION THREAD

**Discussion by SupremeDeity** [fabricjs/fabric.js#10133] [Q&A]
So im trying to send objects forwards and backwards, so they render below or top of other objects, programatically. I have seen quite a bit of issues with this.
1) If i use `canvas.bringObjectForward(obj);`, i have to press the button quite a bit of times before the item actually comes up. IIRC, using true to intersection does not do anything either. There is some change but i do not know what, because it looks like the stroke gets a bit sharper, but i assume thats just it being re-rendered or something with Proxy(). My shapes are a bit blurry right now and doing zoom in zoom out can do something similar.
2) Bringing the object to front adds a `Proxy()` wrapper to the object. Sending it to the back again, removes this wrapper:
![image](https://github.com/user-attachments/assets/c0dd170a-3cbb-46cc-9a23-389103fd6158)

3) Cannot send objects to the back at all unless the other object is wrapped with this `Proxy()`, basically i need to influence the object with `bringObjectForward`

4) Couldn't find the condition for this but with one of the functions, iirc the ToFront and ToBack variants were cloning the object or something, with one of them on top and the other in the same layer in the stack.

This is breaking my history functionality too because structuredClone isnt working with the window property or something. Additional bit of context here. The object being sent to these functions and the output you are seeing in the screenshot all are taken from `getActiveObjects()`. These are custom shapes (subclassed).  This is how the `_render` function goes:
```js
 _render(ctx) {
        ctx.save();
        ctx.translate(-this.width / 2, -this.height / 2);

        const roughCanvas = rough.canvas(ctx.canvas);
        roughCanvas.draw(this.roughDiamond);

        ctx.restore();
    }
```

**Comment by asturur** [comment]
I m not sure what this `Proxy()` is and why you think it should be there.
There are no Proxy() in fabricJS.
So is either something of your code or something else adding it.

BringObjectForward with intersecting should work, if it doesn't you should get a working reproduction issue and open an issue.
BringObjectToTop should always work without blurrying anything.

**Reply by SupremeDeity** [reply]
I will try getting a jsfiddle up when i get the time. 

Proxy() isnt something from my codebase nor i think it should be there. Its just something that's getting added in the middle somehow when i send objects to the back. Also the console's screenshot that i showed is printed directly from getActiveObjects on a mouse event without any other changes. I have never seen this happen before and its only happening when the object gets sent to the back. 

I will take another look at the codebase to see if its being added indirectly somehow. I feel like its most likely happening from cloning or something, since proxy() is something native to js.
